components/brave_wallet/common/brave_wallet.mojom

```diff
@@ -687,11 +689,26 @@ enum TransactionType {
   Other
 };
 
+// TODO(jocelyn): Fill the struct with needed fields.
+struct SolanaTxData {
+};
+
+// TODO(spylogsster): Fill the struct with needed fields.
+struct FilTxData {
+};
+
+union TxDataUnion {
+  TxData eth_tx_data;
+  TxData1559 eth_tx_data_1559;
+  SolanaTxData solana_tx_data;
+  FilTxData fil_tx_data;
+};
```

Comment:
On second thought, this will block the PR :(

Reply:
Yea it definitely is, this PR is based on the union change in another PR, won't merge this until iOS support is added.